src/resolve/resolved-varlink.c

```diff
@@ -15,11 +15,27 @@ typedef struct LookupParameters {
         char *name;
 } LookupParameters;
 
+typedef struct LookupParametersResolveService {
+        char *name;
+        char *type;
+        char *domain;
```

Comment:
as mentioned further down, please use `const char*` here, and fill in the fields via json_dispatch_const_string() as then you don't need to copy them, and also not free them